lib/spack/spack/cmd/__init__.py

```diff
@@ -109,15 +106,19 @@ def parse_specs(args, **kwargs):
             elif normalize:
                 spec.normalize()
 
-        return specs
-
     except spack.parse.ParseError as e:
-        tty.error(e.message, e.string, e.pos * " " + "^")
-        sys.exit(1)
+        tty.die(e.message, e.string, e.pos * " " + "^")
 
     except spack.spec.SpecError as e:
-        tty.error(e.message)
-        sys.exit(1)
+        tty.die(e.message)
+
+    if allow_multi:
+        return specs
+
+    if len(specs) != 1:
+        tty.die('only one spec can be installed at a time.')
+    spec = specs.pop()
```

Comment:
The docstring for `parse_specs()` indicates that it is a "convenience function."  That means (to me) that it's not really necessary, but its entire purpose is to factor stuff out of places that use it, making other code neater and easier to deal with.  Adding the `allow_multi` functionality to `parse_specs()` does exactly that, allowing the following lines to be factored out of at least 2 places:
```
    if len(specs) != 1:
        tty.die('only one spec can be installed at a time.')
    spec = specs.pop()
    return spec
```

Point taken on the error message.

> The fact that currently it is not is a bug (see #1248 and #1976).

Bug or no bug, I took existing code and  refactored it into a convenience function.